libs/langchain/langchain/document_loaders/batch_loader.py

```diff
@@ -0,0 +1,252 @@
+            raise ImportError(
+                "You must install tqdm to use show_progress=True."
+                "You can install tqdm with `pip install tqdm`."
+            ) from err
+        return list(tqdm(iterable, total=args_length))
+    return list(iterable)
+
+
+class BatchLoader(BaseLoader):
+    """
+    Meta loader use to load multiple data from a loader either in parallel
+    or sequentially.
+
+    Example:
+
+    Using a class that is instantiate with __init__
+    ```
+    file_path_list : List[str] = [...]
+    loader_files = BatchLoader(
+        TextLoader,
+        {
+            "file_path": file_path_list
+        },
+    )
+    ```
+    Using a static method to initialize the loader:
+    ```
+    youtube_video_urls : List[str] = [...]
+    loader_youtube = BatchLoader(
+        YoutubeLoader.from_url,
+        {
+            "url": youtube_video_urls
+        },
+    )
+    ```
+
+    Then you can load your data:
+    ```
+    documents = loader_files.load(method="thread", max_workers=4)
+    documents = loader_youtube.load(method="process", max_workers=4)
+    documents = loader_youtube.load(method="sequential")
+    documents = loader_youtube.load(method="async")
+    ```
+    """
+
+    def __init__(
+        self,
+        loader_initializer_callable: Callable[..., BaseLoader],
+        init_loader_args: Dict[str, List[Any]],
+        show_progress: bool = False,
+    ):
+        """Initialize the loader with a loader_callable (either a class or a
+        function used to initialize the loader) and a list of loader_args.
+
+        Args:
+            loader_initializer_callable (Callable[..., BaseLoader]): callable
+                used to initialize the loader (class or init function )
+            init_loader_args (Dict[str, List[Any]]): list of loader_args used
+                to initialize the loader. The keys of the dict are the name of
+                the arguments and the values are the list of values for each.
+            show_progress: If true, will show a progress bar as the files are
+                loaded. This forces an iteration through all matching files to
+                count them prior to loading them.
+
+        Raises:
+            ValueError: if the length of the lists in loader_args are not equal
+        """
+        loader_args = list(zip(*init_loader_args.values()))
+        self.loaders_args = [
+            dict(zip(init_loader_args.keys(), loader_arg)) for loader_arg in loader_args
+        ]
+        self.loader_callable = loader_initializer_callable
+        self.show_progress = show_progress
+        self._partial_load = partial(
+            self._load_loader, loader_callable=self.loader_callable
+        )
+
+        if len(set(len(arg) for arg in init_loader_args.values())) != 1:
+            raise ValueError("All loader_args lists must have the same length")
+
+    def load(self, method: str = "sequential", max_workers: int = 1) -> List[Document]:
```

Comment:
The convention right now is to provide all configuration up-front in the the __init__

Reply:
Done in last commit